src/unix/fs.c

```diff
@@ -389,28 +388,17 @@ static ssize_t uv__fs_scandir(uv_fs_t* req) {
   /* NOTE: We will use nbufs as an index field */
   req->nbufs = 0;
 
-  if (n == 0)
-    goto out; /* osx still needs to deallocate some memory */
-  else if (n == -1)
+  if (n == 0) {
+    /* osx still needs to deallocate some memory */
+    if (dents) {
+      free(dents);
```

Comment:
Can you add the `/* Memory was allocated using the system allocator, so use free() here. */` comment here?